tools/inference_engine/include/inference_engine_tensor.h

```diff
@@ -17,12 +17,20 @@ extern "C" {
         XESS_DATA_TYPE_UNKNOWN = -1000,
     } inference_engine_data_type_t;
 
-typedef struct _inference_engine_tensor_t
-{
-    inference_engine_data_type_t data_type;
-    uint64_t dims[INFERENCE_ENGINE_MAX_TENSOR_DIMS];
-    uint64_t strides[INFERENCE_ENGINE_MAX_TENSOR_DIMS];
-} inference_engine_tensor_t;
+	typedef enum {
+		TENSOR_LAYOUT_NCHW,
+		TENSOR_LAYOUT_NHWC,
+		// .. others... 
+		TENSOR_LAYOUT_UNDEFINED = INT32_MAX
+	} inference_engine_tensor_layout_t;
```

Comment:
I'm not sure if we really need enum for a layout. Currently we expose "dims" and "strides" which are enough to build any plain layout. For block layouts we may need to add another entry "blocks"